target/linux/ipq806x/files/arch/arm/boot/dts/qcom-ipq8064-rb3011uias.dts

```diff
@@ -0,0 +1,599 @@
+
+		act: act {
+			label = "rb3011uias:green:act";
+			gpios = <&qcom_pinmux 33 GPIO_ACTIVE_HIGH>;
+		};
+	};
+
+	sfp_gmac1: sfp {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c_sfp>;
+		los-gpios = <&qcom_pinmux 68 GPIO_ACTIVE_HIGH>;
+		maximum-power-milliwatt = <2000>;
+		mod-def0-gpios = <&qcom_pinmux 63 GPIO_ACTIVE_LOW>;
+		pinctrl-0 = <&sfp_pins>;
+		pinctrl-names = "default";
+//		rate-select0-gpios = <&qcom_pinmux 50 GPIO_ACTIVE_HIGH>;
+//		rate-select0-gpios = <&qcom_pinmux 50 GPIO_ACTIVE_LOW>;
+//		tx-disable-gpios = <&qcom_pinmux 49 GPIO_ACTIVE_HIGH>;
+		tx-disable-gpios = <&qcom_pinmux 49 GPIO_ACTIVE_LOW>;
+		tx-fault-gpios = <&qcom_pinmux 67 GPIO_ACTIVE_HIGH>;
+	};
+};
+
+&adm_dma {
+	status = "okay";
+};
+
+&soc {
+	mdio1: mdio@37200000 {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		compatible = "qcom,ipq8064-mdio", "syscon";
+		reg = <0x37200000 0x200000>;
+		resets = <&gcc GMAC_CORE2_RESET>;
+		reset-names = "stmmaceth";
+		clocks = <&gcc GMAC_CORE2_CLK>;
+		clock-names = "stmmaceth";
+
+		status = "disabled";
+	};
+};
+
+&qcom_pinmux {
+	pinctrl-0 = <&port10_poe_pin>;
+	pinctrl-names = "default";
+
+	led_pins: led-pins {
+		mux {
+			pins = "gpio33";
+			function = "gpio";
+			drive-strength = <2>;
+			bias-pull-down;
+		};
+	};
+
+	mdio1_pins: mdio1-pins {
+		mux {
+			pins = "gpio10", "gpio11";
+			function = "mdio";
+			drive-strength = <8>;
+			bias-disable;
+		};
+	};
+
+	reset_button_pins: reset-button-pins {
+		mux {
+			pins = "gpio66";
+			function = "gpio";
+			drive-strength = <2>;
+			bias-pull-up;
+		};
+	};
+
+	sfp_pins: sfp-pins {
+		mux {
+			pins = "gpio49", "gpio50", "gpio63", "gpio67", "gpio68";
+			function = "gpio";
+			drive-strength = <16>;
+			bias-disable;
+		};
+	};
+
+	sw0_and_phy_reset_pin: switch0-reset-pin {
+		mux {
+			pins = "gpio16";
+			function = "gpio";
+			drive-strength = <16>;
+			bias-pull-down;
+			output-high;
+		};
+	};
+
+	sw1_and_phy_reset_pin: switch1-reset-pin {
+		mux {
+			pins = "gpio17";
+			function = "gpio";
+			drive-strength = <16>;
+			bias-pull-down;
+			output-high;
+		};
+	};
+
+	usb0_pwr_en_pins: usb0-pwr-en-pins {
+		mux {
+			pins = "gpio4";
+			function = "gpio";
+			drive-strength = <16>;
+			bias-pull-down;
+			output-high;
+		};
+	};
+
+	port10_poe_pin: port10-poe-pin {
+		mux {
+			pins = "gpio26";
+			function = "gpio";
+			drive-strength = <16>;
+			bias-pull-down;
+			output-high;
+		};
+	};
+
+};
+
+&gsbi5 {
+	qcom,mode = <GSBI_PROT_SPI>;
+	status = "okay";
+
+	spi@1a280000 {
+		status = "okay";
+
+		pinctrl-0 = <&spi_pins>;
+		pinctrl-names = "default";
+
+		cs-gpios = <&qcom_pinmux 20 GPIO_ACTIVE_HIGH>;
+
+		flash: m25p80@0 {
+			compatible = "jedec,spi-nor";
+			#address-cells = <1>;
+			#size-cells = <1>;
+			spi-max-frequency = <51200000>;
+			reg = <0>;
+
+			partitions {
+				compatible = "fixed-partitions";
+				#address-cells = <1>;
+				#size-cells = <1>;
+
+				SBL1@0 {
+					label = "SBL1";
+					reg = <0x00000000 0x20000>;
```

Comment:
This is probably easier to read if you use 6 digits for all numbers here (offset and size). Start may be 0x0.